src/app/component/MyAccountCustomerForm/MyAccountCustomerForm.component.js

```diff
@@ -45,6 +46,27 @@ export class MyAccountCustomerForm extends FieldForm {
             lastname: {
                 label: __('Last name'),
                 validation: ['notEmpty']
+            },
+            ...this.getVatField()
+        };
+    }
+
+    getVatField() {
+        const { showTaxVatNumber } = this.props;
+        const validation = [];
+
+        if (!showTaxVatNumber) {
+            return {};
+        }
+
+        if (showTaxVatNumber === SHOW_VAT_NUMBER_REQUIRED) {
+            validation.push('notEmpty');
+        }
```

Comment:
I saw this logic... I do not think, however we should resolve this.

Reply:
Done.